### DISCUSSION TITLE
Duplicate Entry Button

### DISCUSSION THREAD

**Discussion by CrispyCrafter** [aminalaee/sqladmin#370] [Ideas]
We've been using SQLAdmin internally to manage a large fleet of IoT devices. 
One pain point at the moment is the inability to duplicate fields. 

Typically a batch of devices would be provisioned in the same way, with minor differences to device IDs

How would one go about including a duplicate button, inline with view and edit?

**Comment by aminalaee** [comment]
Hey, that looks like a common use case.
Let me take a look at other projects like Django/Flask admin to see how they might have done it.
Otherwise we can look into a custom approach,

**Reply by aminalaee** [reply]
https://docs.djangoproject.com/en/4.1/ref/contrib/admin/#django.contrib.admin.ModelAdmin.save_as

**Reply by CrispyCrafter** [reply]
Makes sense that you can simply save-as

**Reply by aminalaee** [reply]
Looks like flask-admin doesn't have this option, so you have to write a custom action.
Anyway I think this is a nice feature, you can duplicate it and edit the fields you want to be different.
Thanks for the suggestion. I will work on it soon.

**Reply by aminalaee** [reply]
@CrispyCrafter Would be great if you could check out the unreleased version, this is already implemented.

**Reply by CrispyCrafter** [reply]
I'll pull this in tommorow and get back to you!
